### ISSUE TITLE
Display is not as advertised

### ISSUE THREAD

**Issue by jwj61** [LMFDB/lmfdb#5040]
On https://beta.lmfdb.org/ModularForm/GL2/Q/holomorphic/1/118/a/a/ you get a link in the q-expansion to display 100 coefficients.  If you click on it, you don't get 100 coefficients.  Instead, you get the 99th coefficient instead of the 9th because it is compressed.

**Comment by jwj61** [comment]
I don't think it should say it will display something if it isn't going to display it.  Maybe it should be a link to copy the expansion through q^99.

**Comment by roed314** [comment]
I think the current behavior is fine.  I think saying "Display 99 coefficients" would look weird, as would having the precision be 101.  Besides, technically we're displaying 100 coefficients, it's just that the coefficient of q^0 is always 0 (note that we are including other 0 coefficients in the count even though the corresponding terms are not shown; for example the coefficient of q^4 is 0 [here](https://beta.lmfdb.org/ModularForm/GL2/Q/holomorphic/23/1/b/a/)).

**Comment by jwj61** [comment]
This is not a question of off by one in the count, or of coefficients which are 0 being suppressed.  The user clicks to display 100 coefficients but it does not display all non-zero terms up to q^99, it only displays a couple of them.

**Comment by roed314** [comment]
Oh, I see.  Yeah, I find that a bit jarring as well.

**Comment by AndrewVSutherland** [comment]
Hmm, this is a bit tricky because we don't really know whether the q-expansion is going to be compressed or not.  Like David I don't think the "100 coefficients" part should change, but I agree that perhaps the word "Display" should change.  Maybe "Get" instead of "Display"?  If the user moves their mouse over the q-expansion at all (which I think someone unhappy with the dots would be likely to do, e.g. to try clicking on them), they will see the copy to clipboard and eyeball icons pop up, both of which will give them access to 100 coefficients of the q-expansion.

**Comment by jwj61** [comment]
That would be a fix.  I may be the only one who thinks this, but I think is ok to have "Display 100 coefficients" and if the user clicks on that, they get what they asked for (without compression).

**Comment by assaferan** [comment]
@jwj61, @AndrewVSutherland, @roed314 - have we come to a conclusion about the wanted behavior in this case? 
I'm fine with our current (compressed) behavior.

**Comment by roed314** [comment]
I agree with @jwj61 that our current behavior is weird.  Here are some alternatives (some of which have been suggested already):
1. When the user clicks "Display 100 coefficients," display them all, uncompressed.  User gets what they ask for.
2. Change "Display" to "Get," but leave the result compressed.  I don't think this improves the situation much.
3. Since we have the ability to compress the display dynamically, the whole Display 10/Display 100 dichotomy is perhaps redundant.  Just always display 100, compressed as normal, and available for copying or viewing via the raw typeset mechanisms.
4. Add more toggle options: start with 10 compressed with a button saying "Display 100 coefficients (compressed)."  When this is clicked, change it to "Display 100 coefficients (uncompressed)" and show them all.

I also think that the raw typeset should include the `+ O(q^100)` (I think we're currently reusing code for polynomials, so it's not available).

**Comment by roed314** [comment]
I think my preference would be option 3.

**Comment by jwj61** [comment]
3 sounds good to me as well.

**Comment by assaferan** [comment]
I also agree with 3.

**Comment by assaferan** [comment]
Solved in PR #6269
